tests/test_idpi/test_time_ops.py

```diff
@@ -0,0 +1,43 @@
+# Standard library
+from pathlib import Path
+
+# Third-party
+import numpy as np
+import pytest
+import xarray as xr
+from numpy.testing import assert_allclose
+
+# First-party
+from idpi import grib_decoder
+from idpi.operators.time_operators import delta
+
+
+@pytest.fixture
+def data_dir():
+    return Path("/project/s83c/rz+/icon_data_processing_incubator/data/temporal")
+
+
+def test_delta(data_dir, fieldextra):
+    steps = np.arange(0, 16, 3)
+    dd, hh = np.divmod(steps, 24)
+    datafiles = [data_dir / f"lfff{d:02d}{h:02d}0000" for d, h in zip(dd, hh)]
+
+    ds = grib_decoder.load_cosmo_data(["TOT_PREC"], datafiles, ref_param="TOT_PREC")
+
+    tot_prec_03h = delta(ds["TOT_PREC"], np.timedelta64(3, "h"))
+    cond = np.logical_or(tot_prec_03h > 0.0, tot_prec_03h.isnull())
+    observed = tot_prec_03h.where(cond, 0.0)
```

Comment:
For the reader who is unfamiliar with fieldextra, one should maybe add a comment that fieldextra sets a missing value if the delta operation cannot be computed, and that in the reference product the result of the computation is explicitly clipped to avoid negative values due to numerical inaccuracy.

Reply:
Why are negative values assumed to be result of numerical inaccuracy? Is it due to the nature of the field, i.e. precipation can not be negative?

Reply:
PS: Negative values in time differences of precipitation may also occur due to the GRIB compression.